lms/resources/_js_config/file_picker_config.py

```diff
@@ -105,10 +105,10 @@ def microsoft_onedrive(cls, request, application_instance):
         }
 
     @classmethod
-    def vital_source_config(cls, _request, application_instance):
+    def vital_source_config(cls, request, _application_instance):
         """Get Vital Source config."""
-        enabled = application_instance.settings.get("vitalsource", "enabled", False)
-        return {"enabled": enabled}
+
+        return {"enabled": request.find_service(VitalSourceService).enabled}
```

Comment:
It's not enough to know that the tick box is on. You also need to check all the required fields exist.